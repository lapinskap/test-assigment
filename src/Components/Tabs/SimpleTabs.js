import PropTypes from 'prop-types';
import React, { useContext } from 'react';

import Tabs, { TabPane } from 'rc-tabs';
import TabContent from 'rc-tabs/lib/TabContent';
import DefaultHashTabBar from './DefaultHashTabBar';
import { filterTabsByAcl } from '../../utils/RoleBasedSecurity/filters';
import RbsContext from '../../utils/RoleBasedSecurity/RbsContext';

export default function SimpleTabs({
  tabsConfig = [], activeKey, defaultActiveKey, animated = false, destroyInactiveTabPane = true,
}) {
  const { userInfo } = useContext(RbsContext);
  const currentKey = activeKey || defaultActiveKey;
  return (
    <Tabs
      animated={animated}
      activeKey={currentKey}
      destroyInactiveTabPane={destroyInactiveTabPane}
      renderTabBar={() => <DefaultHashTabBar />}
      renderTabContent={() => <TabContent animated={animated} />}
    >
      {filterTabsByAcl(tabsConfig, userInfo).map(({
        name, key, component, disabled, tabClassName = null, tabTitle = null,
      }) => (
        <TabPane
          tab={<span className={tabClassName} title={tabTitle}>{name}</span>}
          key={key}
          disabled={Boolean(disabled)}
        >
          {component}
        </TabPane>
      ))}
    </Tabs>
  );
}

SimpleTabs.propTypes = {
  activeKey: PropTypes.string,
  animated: PropTypes.bool,
  destroyInactiveTabPane: PropTypes.bool,
  defaultActiveKey: PropTypes.string.isRequired,
  tabsConfig: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    key: PropTypes.string,
    component: PropTypes.node,
    disabled: PropTypes.bool,
  })).isRequired,
};

SimpleTabs.defaultProps = {
  activeKey: '',
  animated: false,
  destroyInactiveTabPane: true,
};
