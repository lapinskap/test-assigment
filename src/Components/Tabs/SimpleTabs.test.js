/* eslint-disable function-call-argument-newline */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from 'react';
import renderer from 'react-test-renderer';
import SimpleTabs from './SimpleTabs';
import '../../tests/setupTests';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    location: {
      hash: 'test',
    },
  }),
  Prompt: ({ message }) => <div>{message}</div>,
  Link: ({ to, children, ...props }) => <a href={to} {...props}>{children}</a>,
}));

describe('<SimpleTabs />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  let tabsConfig = []; let activeKey; let defaultActiveKey; const animated = false; const
    destroyInactiveTabPane = true;

  beforeEach(() => {
    tabsConfig = [
      {
        key: '20',
        name: 'test',
        icon: 's-glyph-5',
      },
    ];
    activeKey = 'test';
    defaultActiveKey = 'test';
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default state', () => {
    const tree = renderer
      .create(<SimpleTabs
        activeKey={activeKey}
        tabsConfig={tabsConfig}
        defaultActiveKey={defaultActiveKey}
        animated={animated}
        destroyInactiveTabPane={destroyInactiveTabPane}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
