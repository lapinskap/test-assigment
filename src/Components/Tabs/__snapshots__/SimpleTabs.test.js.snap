// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleTabs /> renders correctly with default state 1`] = `
<div
  className="rc-tabs rc-tabs-top"
  onScroll={[Function]}
  style={Object {}}
>
  <div
    className="my-3 card"
  >
    <div
      className="card-body"
    >
      <div
        className="rc-tabs-bar"
        data-t1="tabs"
        role="tablist"
        tabIndex="0"
      >
        <div>
          <a
            className="rc-tabs-tab-link"
            href="#20"
          >
            <div
              aria-disabled="false"
              aria-selected="true"
              className="rc-tabs-tab"
              data-t1="20"
              role="tab"
            >
              <span
                className={null}
                title={null}
              >
                test
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    onKeyDown={[Function]}
    role="presentation"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
        "position": "absolute",
        "width": 0,
      }
    }
    tabIndex={0}
  />
  <div
    className="rc-tabs-content rc-tabs-content-no-animated"
  >
    <div
      aria-hidden="true"
      className="rc-tabs-tabpane rc-tabs-tabpane-inactive"
      role="tabpanel"
    />
  </div>
  <div
    onKeyDown={[Function]}
    role="presentation"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
        "position": "absolute",
        "width": 0,
      }
    }
    tabIndex={0}
  />
</div>
`;
